File: AEDS-2/ATIVIDADE/LEITURA-DADOS/leituraDados.java
import java.util.Scanner;
import java.util.Locale;

public class leituraDados {
    public static void processarRegistro(int id, double preco, String tempo, String cidade) {
        System.out.printf("Registro lido: %d %.2f %s %s\n", id, preco, tempo, cidade);
      }

    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        scan.useLocale(Locale.US);

        int id;
        double preco;
        String tempo;
        String cidade;

        while((id = scan.nextInt()) != 0){
            preco = scan.nextDouble();
            tempo = scan.next();
            cidade = scan.nextLine().trim();
            processarRegistro(id, preco, tempo, cidade);
            }
            
        scan.close();
    }
}

File: AEDS-2/ATIVIDADE/LEITURA-DADOS/leituraDados.c
#include <stdio.h>

void processarRegistro(int id, double preco, char* tempo, char* cidade) {
  printf("Registro lido: %d %.2f %s %s\n", id, preco, tempo, cidade);
}

int main(void){
    int id;
    double preco;
    char tempo[6];
    char cidade[50];

    while(scanf(" %d %lf %s %[^\n]", &id, &preco, tempo, cidade) == 4 && id != 0){
        processarRegistro(id, preco, tempo, cidade);   
    }

    return 0;
}